zamba/models/cnnensemble_model.py

```diff
@@ -96,19 +94,41 @@ def predict(self, file_names, resample=True):
                 classes
         """
         processed_paths = self.preprocess_videos(file_names, resample=resample)
-        valid_videos, invalid_videos = zamba.utils.get_valid_videos(processed_paths.values())
-        self.logger.debug(f"Invalid videos {str(invalid_videos)}")
+        valid_videos = list(processed_paths.values())
+        invalid_videos = []
 
         l1_results = {}
+        invalid_mask = np.zeros(len(valid_videos), dtype=bool)
+
         prof = config.PROFILES[self.profile]
         for l1_model in tqdm(prof, desc=f'Predicting on {len(prof)} L1 models'):
-            l1_results[l1_model] = generate_prediction_test(
+            l1_model_preds = generate_prediction_test(
                 model_name=l1_model,
                 weights=config.MODEL_DIR / 'output' / config.MODEL_WEIGHTS[l1_model],
                 file_names=valid_videos,
                 save_results=False
             )
 
+            l1_results[l1_model] = l1_model_preds
+
+            # if video is invalid for this model, track that
+            invalid_mask |= np.isnan(l1_model_preds).all(axis=(1, 2))
+
+        # remove invalid videos from l1 results:
+        l1_results = {
+            name: results[~invalid_mask, ...]
```

Comment:
what does `...` do here? is that a shortcut for selecting all other dimensions?

Reply:
Yep `...` gets all the other dimensions.